boards/nordic/nrf51dk/Makefile

```diff
@@ -6,22 +6,24 @@ PLATFORM=nrf51dk
 
 include ../../Makefile.common
 
-.PHONY: apps/$(APP)/build/$(TOCK_ARCH)/app.bin
-apps/$(APP)/build/$(TOCK_ARCH)/app.bin:
-	@make -C apps/$(APP) TOCK_ARCH=$(TOCK_ARCH)
+TOCKLOADER=tockloader
 
-target/$(TARGET)/release/nrf51dk-$(APP).hex: target/$(TARGET)/release/nrf51dk-$(APP)
-	@$(OBJCOPY) -Oihex $^ $@
+# Where in the nrf51 flash to load the kernel with `tockloader`
+KERNEL_ADDRESS=0x00000
 
-JLINK=JLinkExe
-JLINK_OPTIONS+=-device nrf51422 -if swd -speed 1200 -AutoConnect 1
-JLINK_SCRIPTS_DIR=jtag/
+# Upload programs over uart with tockloader
+ifdef PORT
+  TOCKLOADER_GENERAL_FLAGS += --port $(PORT)
+endif
```

Comment:
Should we make this general / included as an option on all boards?

Reply:
I vote no.

I'm personally fairly resistant to how much board specific stuff is getting pushed into the the common Makefile. It really makes maintaining a separate board without checking Tock out as a subrepository just for the Makefiles kind of un-tennable (and I think using LLVM's tools that come with Rust will help with that significantly). There aren't other boards in upstream that use JLink anyway.

Reply:
Is there a lot of board-specific content in the makefile? `PLATFORM` and `TARGET` are passed in as variables, but otherwise it is generic assuming rust can compile to your target and objcopy can create a valid binary file.

Reply:
It's not terrible right now, though it's still more complex than I would like (again, I suspect switching to LLVM will let us simplify the common Makefile significantly).

Adding `tockloader` stuff to the common Makefile feels board specific to me---not all boards will use tockloader, and certainly most early ones won't. If tockloader is that simple to use (I think it's either there or getting there for the boards it supports), it should be at least as easy and readable to write the one liner `tockloader` rule directly in the board Makefile as it is to rely on a particular structure and API that the common Makefile exports.

That's all. And it's really just a feeling. Like, had the PR added this option to the common Makefile instead of the board Makefile I likely wouldn't have objected (if I even noticed).